### DISCUSSION TITLE
Q: what is the purpose again of `DynamicEventGridAuthorizationAttribute` and may we improve it?

### DISCUSSION THREAD

**Discussion by stijnmoreels** [arcus-azure/arcus.eventgrid#164] [Q&A] [answered]
We have this attribute called `DynamicEventGridAuthorizationAttribute` with a `static` public property to inject(?) a secret return value on some parts of an application.
Can/should/may we improve this to a more useful and production-ready approach?

**Comment by tomkerkhove** [comment, marked as answer]
I believe this is something @SamVanhoutte has contributed.

**Reply by stijnmoreels** [reply]
Ok, I've read the feature docs about this attribute. Seems like there's been taking a shortcut by using this public static property to retrieve a secret key. We should probably use a different approach where we reference an authentication filter in the attribute that takes in an `ISecretProvider` instead of hardcoding the secret.

**Reply by SamVanhoutte** [reply]
I believe that was from the initial starting pilot code (though it probably was named differently).  The intention was to declare on operation level how the secrets (name + value) should be retrieved.  `ISecretProvider` definitely looks like the way to go.  So making this `Obsolete` might be a good choice.
